ui-ngx/src/app/shared/components/time/timewindow-panel.component.ts

```diff
@@ -78,107 +80,95 @@ export class TimewindowPanelComponent extends PageComponent implements OnInit {
-              }),
-            interval: [
-              this.timewindow.realtime && typeof this.timewindow.realtime.interval !== 'undefined'
-                ? this.timewindow.realtime.interval : null
-            ],
-            quickInterval: this.fb.control({
-              value: this.timewindow.realtime && typeof this.timewindow.realtime.quickInterval !== 'undefined'
-                ? this.timewindow.realtime.quickInterval : null,
-              disabled: hideInterval || hideQuickInterval
-            })
-          }
-        ),
-        history: this.fb.group(
-          {
-            historyType: this.fb.control({
-              value: this.timewindow.history && typeof this.timewindow.history.historyType !== 'undefined'
-                ? this.timewindow.history.historyType : HistoryWindowType.LAST_INTERVAL,
-              disabled: hideInterval
-            }),
-            timewindowMs: this.fb.control({
-              value: this.timewindow.history && typeof this.timewindow.history.timewindowMs !== 'undefined'
-                ? this.timewindow.history.timewindowMs : null,
-              disabled: hideInterval
-            }),
-            interval: [
-              this.timewindow.history && typeof this.timewindow.history.interval !== 'undefined'
-                ? this.timewindow.history.interval : null
-            ],
-            fixedTimewindow: this.fb.control({
-              value: this.timewindow.history && typeof this.timewindow.history.fixedTimewindow !== 'undefined'
-                ? this.timewindow.history.fixedTimewindow : null,
-              disabled: hideInterval
-            }),
-            quickInterval: this.fb.control({
-              value: this.timewindow.history && typeof this.timewindow.history.quickInterval !== 'undefined'
-                ? this.timewindow.history.quickInterval : null,
-              disabled: hideInterval
-            })
-          }
-        ),
-        aggregation: this.fb.group(
-          {
-            type: this.fb.control({
-              value: this.timewindow.aggregation && typeof this.timewindow.aggregation.type !== 'undefined'
-                ? this.timewindow.aggregation.type : null,
-              disabled: hideAggregation
-            }),
-            limit: this.fb.control({
-              value: this.timewindow.aggregation && typeof this.timewindow.aggregation.limit !== 'undefined'
-                ? this.checkLimit(this.timewindow.aggregation.limit) : null,
-              disabled: hideAggInterval
-            }, [])
-          }
-        ),
-        timezone: this.fb.control({
-          value: this.timewindow.timezone !== 'undefined'
-            ? this.timewindow.timezone : null,
-          disabled: hideTimezone
-        })
+      realtime: this.fb.group({
+        realtimeType: [{
+          value: this.defined(realtime, realtime.realtimeType) ? this.timewindow.realtime.realtimeType : RealtimeWindowType.LAST_INTERVAL,
```

Comment:
It may be better to replace it with a function from the product code - **isDefined**. Example: `IsDefined(realtime?.realtimeType) `